std/traits.d

```diff
@@ -2513,11 +2513,8 @@ Returns:
 template isInnerClass(T)
 if (is(T == class))
 {
-    import std.meta : staticIndexOf;
-
     static if (is(typeof(T.outer)))
-        enum isInnerClass = __traits(isSame, typeof(T.outer), __traits(parent, T))
-                         && (staticIndexOf!(__traits(allMembers, T), "outer") == -1);
```

Comment:
`is(typeof(T.outer))` already checks if there's an `outer` member, which is what `staticIndexOf` does again.

Reply:
Related question: any reason to prefer `is(...)` to `__traits(hasMember, T, "outer")` ?

Reply:
Isn't the `staticIndexOf... == -1` verifying that there is *not* a member named "outer"?

Reply:
You are right. That is what I was trying to say, but I had a poor wording.

Now actually, I'm not sure what's supposed to be doing, because [staticIndexOf](https://dlang.org/phobos/std_meta.html#.staticIndexOf) checks if the first template argument is in the remainder of the template arguments list.

If it does indeed check if there is *no* `outer` member, won't the entire check evaluate to `false`, since it's part of the `is(typeof(T.outer))` branch path?

Reply:
The arguments of `staticIndexOf` are REVERSED and that's a bug that should be caught in unittesting. @edi33416 can you please do that in this pr?

Reply:
Can use `!hasMember!(T, "outer")`, see https://dlang.org/library/std/traits/has_member.html

Reply:
using `__traits` directly will in most cases be faster than going through a template

Reply:
I tried this suggestion; it didn't work.
`__traits(hasMember, T, "outer") == true` for inner classes... even though it's not an implicit member.
I went with this: https://github.com/dlang/druntime/pull/2398/files#diff-f1322962163674c1ee8c7a018193ab45R160
It works.